chainerx_cc/chainerx/routines/math.cc

```diff
@@ -878,6 +879,72 @@ Array Sqrt(const Array& x) {
     return out;
 }
 
+void PowerImpl(const Array& x1, const Array& x2, const Array& out) {
+    {
+        NoBackpropModeScope scope{};
+        x1.device().backend().CallKernel<PowerKernel>(x1, x2, out);
+    }
+
+    BackwardBuilder bb{"power", {x1, x2}, out};
+    if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+        bt.Define([x1_tok = bb.RetainInput(0), x2_tok = bb.RetainInput(1)](BackwardContext& bctx) {
+            const Array& x1 = bctx.GetRetainedInput(x1_tok);
+            const Array& x2 = bctx.GetRetainedInput(x2_tok);
+            const Array& gin = *bctx.output_grad() * x2 * Power(x1, x2 - Scalar{1, GetKind(x2.dtype())});
+            bctx.input_grad() = x1.dtype() != gin.dtype() ? gin.AsType(x1.dtype()) : gin;
+        });
+    }
+    if (BackwardBuilder::Target bt = bb.CreateTarget(1)) {
+        bt.Define([x1_tok = bb.RetainInput(0), out_tok = bb.RetainOutput(0), x2_dtype = x2.dtype()](BackwardContext& bctx) {
+            const Array& x1 = bctx.GetRetainedInput(x1_tok);
+            const Array& out = bctx.GetRetainedOutput(out_tok);
+            const Array& gin = *bctx.output_grad() * out * Log(x1);
+            bctx.input_grad() = x2_dtype != gin.dtype() ? gin.AsType(x2_dtype) : gin;
+        });
+    }
+    bb.Finalize();
+}
+
+void PowerASImpl(const Array& x1, Scalar x2, const Array& out) {
```

Comment:
Is this behavior intended?
```py
>>> chainerx.power(chainerx.array([2]), -1)
array([0], shape=(1,), dtype=int64, device='native:0')
```
How about raising an error in such a case?

Reply:
Discussed to leave it as is. Similar to e.g. CuPy.